cparser/Elab.ml

```diff
@@ -1439,9 +1470,8 @@ let elab_expr vararg loc env a =
       let ty = match b3.edesc with ESizeof ty -> ty | _ -> assert false in
       let ty' = default_argument_conversion env ty in
       if not (compatible_types AttrIgnoreTop env ty ty') then
-        warning "'%a' is promoted to '%a' when passed through '...'.@ You should pass '%a', not '%a', to 'va_arg'"
-                Cprint.typ ty Cprint.typ ty'
-                Cprint.typ ty' Cprint.typ ty;
+        warning Varargs "second argument to 'va_arg' is of promotable type %a; this va_arg has undefined behavior because arguments will be promoted to %a"
+          (print_typ env) ty (print_typ env) ty';
```

Comment:
I still want either GCC's or CompCert's original message instead of clang's.  The former two tell the programmers what to do to fix their mistake.  clang's message is just obscure.